src/hooks/useFiatText.ts

```diff
@@ -80,8 +80,12 @@ export const formatFiatString = (props: {
 }): string => {
   const { fiatAmount, minPrecision = 2, maxPrecision = 6, autoPrecision = true, noGrouping = false } = props
 
+  // Assume any spaces means this is some truncated '1.23 Bn' or '3.45 M' string
+  const magnitudeCode = fiatAmount.includes(' ') ? ` ${fiatAmount.split(' ')[1]}` : ''
+  const fiatAmountCleanedMag = magnitudeCode.length === 0 ? fiatAmount : fiatAmount.split(' ')[0]
+
```

Comment:
Since this routine runs in A LOT of places in the app, I'm going to be nitpicky on performance.

Optimize the operations 

```
const [fiatAmountCleanedMag, magnitudeCode] = fiatAmount.split(' ')
const magnitudeCodeStr = magnitudeCode == null ? '' : ` ${magnitudeCode}`
```

Then use `magnitudeCodeStr` in the return value